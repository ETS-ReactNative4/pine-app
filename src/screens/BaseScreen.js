import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import ErrorBoundary from '../components/ErrorBoundary';
import BackHeaderContainer from '../containers/BackHeaderContainer';

const styles = StyleSheet.create({
  view: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ffffff',
    padding: 40
  }
});

export default class BaseScreen extends Component {
  renderHeader() {
    if (this.props.hideHeader) {
      return null;
    }

    return (
      <BackHeaderContainer
        backButtonIconStyle={this.props.backButtonIconStyle}
        title={this.props.headerTitle}
      />
    );
  }

  render() {
    return (
      <ErrorBoundary {...this.props} style={[styles.view, this.props.style]}>
        {this.renderHeader()}
        {this.props.children}
      </ErrorBoundary>
    );
  }
}

BaseScreen.propTypes = {
  style: PropTypes.any,
  backButtonIconStyle: PropTypes.any,
  headerTitle: PropTypes.string,
  hideHeader: PropTypes.bool,
  children: PropTypes.node
};
