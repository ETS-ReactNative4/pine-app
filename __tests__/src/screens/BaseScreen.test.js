import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BaseScreen from '../../../src/screens/BaseScreen';

jest.mock('../../../src/containers/BackHeaderContainer', () => 'BackHeaderContainer');

describe('BaseScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BaseScreen />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
