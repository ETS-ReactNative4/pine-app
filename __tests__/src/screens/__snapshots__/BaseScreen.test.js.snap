// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseScreen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
        "padding": 40,
      },
      undefined,
    ]
  }
>
  <BackHeaderContainer
    backButtonIconStyle={undefined}
    title={undefined}
  />
  <ErrorModalContainer />
</View>
`;
